From jirislaby@gmail.com Thu Mar 26 01:34:24 2009
From: Jiri Slaby <jirislaby@gmail.com>
Date: Thu, 26 Mar 2009 09:34:12 +0100
Subject: staging: meilhaus, annotate cpi functions
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org, Jiri Slaby <jirislaby@gmail.com>, David Kiliani <mail@davidkiliani.de>
Message-ID: <1238056453-25653-7-git-send-email-jirislaby@gmail.com>


Add __devinit and __devexit to pci probe/remove. Also make pci_driver
static.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Cc: David Kiliani <mail@davidkiliani.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/meilhaus/memain.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/staging/meilhaus/memain.c
+++ b/drivers/staging/meilhaus/memain.c
@@ -103,11 +103,11 @@ static const struct file_operations me_f
 	.release = me_release,
 };
 
-struct pci_driver me_pci_driver = {
+static struct pci_driver me_pci_driver = {
 	.name = MEMAIN_NAME,
 	.id_table = me_pci_table,
 	.probe = me_probe_pci,
-	.remove = me_remove_pci
+	.remove = __devexit_p(me_remove_pci),
 };
 
 /* //me_usb_driver
@@ -384,7 +384,8 @@ static me_device_t *get_dummy_instance(u
 	return instance;
 }
 
-static int me_probe_pci(struct pci_dev *dev, const struct pci_device_id *id)
+static int __devinit me_probe_pci(struct pci_dev *dev,
+		const struct pci_device_id *id)
 {
 	int err;
 	me_pci_constructor_t constructor = NULL;
@@ -582,7 +583,7 @@ static int insert_to_device_list(me_devi
 	return 0;
 }
 
-static void me_remove_pci(struct pci_dev *dev)
+static void __devexit me_remove_pci(struct pci_dev *dev)
 {
 	int vendor_id = dev->vendor;
 	int device_id = dev->device;
